src/drive/web/modules/selectors.js

```diff
@@ -10,3 +13,34 @@ export const getCurrentFolderId = rootState => {
 export const getCurrentFileId = rootState => {
   return rootState.router.params.fileId
 }
+
+const getFolderContentQueries = (rootState, folderId) => {
+  const queries = rootState.cozy.queries
+  const folderContentQueries = Object.entries(queries)
+    .filter(([queryId]) => {
+      const parsed = parseFolderQueryId(queryId)
+      if (!parsed) {
+        return false
+      }
+      const { folderId: queryFolderId } = parsed
+      if (queryFolderId !== folderId) {
+        return false
+      }
+      return true
+    })
+    .map(x => x[1])
+  return folderContentQueries
+}
+
+export const getFolderContent = (rootState, folderId) => {
+  const folderContentQueries = getFolderContentQueries(rootState, folderId)
+  if (folderContentQueries.length > 0) {
+    const mostRecentQueryResults = maxBy(
+      folderContentQueries,
+      x => x.lastUpdate
+    )
+    return mostRecentQueryResults.data
+  } else {
+    return null
+  }
```

Comment:
I though, I don't know exactly how, that this is the job of cozyclient to handle that kind of stuff. Shouldn't we use getDocumentFromState or something? 🤔

Reply:
Mmh I am not sure that something like this should be in cozy-client. Here we use applicative knowledge to know that different queries can be merged as a single query. Maybe we could achieve something similar with a query group (this would be a new notion in cozy-client), but I am not sure that we have clear ideas on this.